sdk/src/trace/always_off_sampler.cc

```diff
@@ -0,0 +1,30 @@
+#include "opentelemetry/sdk/trace/always_off_sampler.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+namespace trace_api = opentelemetry::trace;
+
+SamplingResult AlwaysOffSampler::ShouldSample(
+    const SpanContext *parent_context,
+    trace_api::TraceId trace_id,
+    nostd::string_view name,
+    trace_api::SpanKind span_kind,
+    const nostd::span<AttributeKeyValue> &attributes) noexcept
+  {
+    return
+    {
+      Decision::NOT_RECORD,
+      std::unique_ptr<nostd::span<AttributeKeyValue>>(new nostd::span<AttributeKeyValue>())
```

Comment:
You can make this a `{ Decisions::NOT_RECORD, nullptr }`. I think we should agree on returning a `nullptr` in cases when no attributes are added, as this saves heap spaces and gets rid of an obsolete allocation (for every span).

Reply:
Sounds good